pkg/bundle/controller.go

```diff
@@ -76,29 +133,26 @@ func AddBundleController(ctx context.Context, mgr manager.Manager, opts Options)
 
 	// Only reconcile config maps that match the well known name
 	if err := ctrl.NewControllerManagedBy(mgr).
-		// Reconcile trust.cert-manager.io Bundles
-		For(new(trustapi.Bundle)).
+		Named("bundles").
+
+		////// Targets //////
 
 		// Reconcile over owned ConfigMaps in all Namespaces. Only cache metadata.
 		// These ConfigMaps will be Bundle Targets
-		Owns(new(corev1.ConfigMap), builder.OnlyMetadata).
+		Watches(source.NewKindWithCache(new(corev1.ConfigMap), targetCache), &handler.EnqueueRequestForOwner{
+			OwnerType:    new(trustapi.Bundle),
+			IsController: true,
+		}, builder.OnlyMetadata).
```

Comment:
question: Would it be possible for you to expand on why we want to do caching for targets here, even metadata-only? I'm struggling to think of situations in which it helps!

Reply:
It is not necessary, but this is the easiest way to watch for resource changes.

Reply:
> It is not necessary, but this is the easiest way to watch for resource changes/ obtain an Informer.

From looking more at the code for controller-runtime, I can see that we'd have to go out of our way to _avoid_ using a cache here, so it makes sense that it'd be easier to just use one. That's good by me!

What I don't understand now is if we were to use the following:

```go
Watches(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForOwner{
			OwnerType:    new(trustapi.Bundle),
			IsController: true,
		}, builder.OnlyMetadata).
```

(Or `Owns(new(corev1.ConfigMap), builder.OnlyMetadata).` as before? That seems equivalent?)

Then from the docs, at some point `InjectCache` will be called which will add a cache [initialized by the ControllerManager](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/source#Kind.InjectCache).

Why do we need to define our own target cache here rather than the one initialized by the ControllerManager? What're we gaining from that? Would it be worth adding that reasoning when we define the targetCache?